File: src/main/java/fr/spoutnik87/musicbot_rest/repository/GroupRepository.java
package fr.spoutnik87.musicbot_rest.repository;

import fr.spoutnik87.musicbot_rest.model.Group;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface GroupRepository extends JpaRepository<Group, Long> {

    Optional<Group> findByUuid(String uuid);
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/repository/GroupRepository.kt
package fr.spoutnik87.musicbot_rest.repository

import fr.spoutnik87.musicbot_rest.model.Group
import fr.spoutnik87.musicbot_rest.model.Server
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
interface GroupRepository : JpaRepository<Group, Long> {
    @Transactional(readOnly = true)
    fun findByUuid(uuid: String): Group?
    @Transactional(readOnly = true)
    fun findByUuidAndServer(uuid: String, server: Server): Group?
}